flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/PartitionShuffleDescriptor.java

```diff
@@ -0,0 +1,139 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.shuffle;
+
+import org.apache.flink.runtime.executiongraph.IntermediateResultPartition;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.IntermediateDataSetID;
+import org.apache.flink.runtime.jobgraph.IntermediateResultPartitionID;
+import org.apache.flink.runtime.state.KeyGroupRangeAssignment;
+
+import javax.annotation.Nonnull;
+
+import java.io.Serializable;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/** Partition descriptor for {@link ShuffleMaster} to obtain {@link ShuffleDeploymentDescriptor}. */
+public class PartitionShuffleDescriptor implements Serializable {
+
+	private static final long serialVersionUID = 6343547936086963705L;
+
+	/** The ID of the result this partition belongs to. */
+	@Nonnull
+	private final IntermediateDataSetID resultId;
+
+	/** The ID of the partition. */
+	@Nonnull
+	private final IntermediateResultPartitionID partitionId;
+
+	/** The type of the partition. */
+	@Nonnull
+	private final ResultPartitionType partitionType;
+
+	/** The number of subpartitions. */
+	private final int numberOfSubpartitions;
+
+	/** The maximum parallelism. */
+	private final int maxParallelism;
+
+	/** Connection index to identify this partition of intermediate result. */
+	private final int connectionIndex;
+
+	public PartitionShuffleDescriptor(
+		@Nonnull IntermediateDataSetID resultId,
+		@Nonnull IntermediateResultPartitionID partitionId,
+		@Nonnull ResultPartitionType partitionType,
+		int numberOfSubpartitions,
+		int maxParallelism,
+		int connectionIndex) {
+
+		this.resultId = resultId;
+		this.partitionId = partitionId;
+		this.partitionType = partitionType;
+		this.connectionIndex = connectionIndex;
+
+		KeyGroupRangeAssignment.checkParallelismPreconditions(maxParallelism);
+		checkArgument(numberOfSubpartitions >= 1);
+		this.numberOfSubpartitions = numberOfSubpartitions;
+		this.maxParallelism = maxParallelism;
+	}
+
+	@Nonnull
+	public IntermediateDataSetID getResultId() {
+		return resultId;
+	}
+
+	@Nonnull
+	public IntermediateResultPartitionID getPartitionId() {
+		return partitionId;
+	}
+
+	@Nonnull
+	public ResultPartitionType getPartitionType() {
+		return partitionType;
+	}
+
+	public int getNumberOfSubpartitions() {
+		return numberOfSubpartitions;
+	}
+
+	public int getMaxParallelism() {
+		return maxParallelism;
+	}
+
+	public int getConnectionIndex() {
```

Comment:
Currently it could be only package private. I am not sure whether it could be used outside of package future.

Reply:
it could be potentially used outside of this package in other shuffle implementation. I would leave methods public here because it is basically part of shuffle API then.